### ISSUE TITLE
Add metadata feature to automagically broadcast it

### ISSUE THREAD

**Issue by aowen87** [visit-dav/issues-test#2649]
Until today, I hadn't realized that VisIt does indeed have the ability to broadcast a metadata object. This means a single rank in parallel can construct it and then broadcast it to all ranks in parallel. This is hugely convenient for dealing with parallel in most if not all database plugins. But, its a bit hidden from view. In fact, the *only* plugin that exploits it is Silo. I have been working with MOAB and Mili plugin developers dealing with this issue and we have always stumbled over the issue that its not easy to handle PopMD in parallel whilst being efficient in terms of filesystem interactions but keeping the algorithm simple.

I recommend the following to help ease development of db plugins for parallel...That VisIt *automagically* broadcast metadata (from rank 0) and only avert that if a plugin requests *not* to. This would be a new feature on metadata object. Call it the mdBcastRank. If set to -1, then its off. Otherwise, it specifies a rank to use as broadcaster and will do the broadcast just prior to return from call to PopMD in the avtFileFormat methods. We would have to touch many existing plugins to *turn it off*.

I rate this as high impact because it hugely simplifies db plugin development.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Mark Miller
Original creation: 04/28/2017 02:26 pm
Original update: 05/16/2017 07:01 pm
Ticket number: 2812
